alfalfa_worker/jobs/openstudio/step_run.py

```diff
@@ -1,25 +1,37 @@
 import os
-import socket
 from datetime import datetime, timedelta
-from time import sleep, time
+from typing import Callable
 
-import mlep
+import openstudio
+from pyenergyplus.api import EnergyPlusAPI
 
+from alfalfa_worker.jobs.openstudio.lib.alfalfa_point import AlfalfaPoint
 from alfalfa_worker.jobs.openstudio.lib.variables import Variables
-from alfalfa_worker.jobs.step_run_base import StepRunBase
-from alfalfa_worker.lib.job import (
+from alfalfa_worker.jobs.step_run_process import StepRunProcess
+from alfalfa_worker.lib.enums import PointType
+from alfalfa_worker.lib.job_exception import (
     JobException,
     JobExceptionExternalProcess,
-    message
+    JobExceptionSimulation
 )
+from alfalfa_worker.lib.models import Point
 
 
-class StepRun(StepRunBase):
+def callback_wrapper(func):
+    def wrapped_func(self: "StepRun", state):
+        try:
+            return func(self, state)
+        except Exception:
+            self.catch_exception()
+    return wrapped_func
+
+
+class StepRun(StepRunProcess):
```

Comment:
Now the fun begins! 

I'm pretty sure I'm following the basic idea. Run EnergyPlus in a separate process and use signaling (`multiprocessing.Manager`) to coordinate between the main and the EnergyPlus process. Seems good to me, but makes me slightly sad that this is exactly what Spawn does and we haven't managed to consolidate this. Not your fault though, so what you have here is good. 

The main overarching comment I have is that the hierarchy of code seems a bit unnecessarily complex and maxes out my cognitive ability just a bit. I'm not suggesting a particular change, but just note that I believe the class hierarchy goes....

`StepRun` -> `StepRunProcess` -> `StepRunBase` -> `Job` and then there is `JobMetaClass` supporting `Job`. That's a lot or hierarchy to keep track of.